src/opening-period/form/OpeningPeriodForm.tsx

```diff
@@ -0,0 +1,315 @@
+      setSubmitStatus('succeeded');
+    } catch (err) {
+      setSubmitStatus('error');
+      // eslint-disable-next-line no-console
+      console.error(err); // For debug purposes
+    }
+  };
+
+  return (
+    <>
+      {submitStatus === 'succeeded' && (
+        <SuccessToast
+          dataTestId="opening-period-form-success"
+          label={successTextAndLabel.label}
+          text={successTextAndLabel.text}
+          onClose={(): void => setSubmitStatus('init')}
+        />
+      )}
+      {submitStatus === 'error' && (
+        <ErrorToast
+          dataTestId="opening-period-form-error"
+          label={errorTextAndLabel.label}
+          text={errorTextAndLabel.text}
+          onClose={(): void => setSubmitStatus('init')}
+        />
+      )}
+      <form
+        id={formId}
+        data-test={formId}
+        className="opening-period-form"
+        onSubmit={handleSubmit(onSubmit)}>
+        <section className="form-section">
+          <OpeningPeriodDescription register={register} errors={errors} />
+        </section>
+        <section className="form-section">
+          <h3 className="opening-period-section-title">Ajanjakso</h3>
+          <section className="opening-period-time-period">
+            <Datepicker
+              id="openingPeriodBeginDate"
+              dataTest="openingPeriodBeginDate"
+              labelText="Alkaa"
+              onChange={(value): void => setPeriodBeginDate(value || null)}
+              value={periodBeginDate}
+              registerFn={register}
+              required
+            />
+            <p className="dash-between-begin-and-end-date">—</p>
+            <Datepicker
+              id="openingPeriodEndDate"
+              dataTest="openingPeriodEndDate"
+              labelText="Päättyy"
+              onChange={(value): void => setPeriodEndDate(value || null)}
+              value={periodEndDate}
+              registerFn={register}
+              required
+            />
+          </section>
+        </section>
+        <section className="form-section time-span-group">
+          <div className="time-span-list-container">
+            <h3 className="opening-period-section-title">Aukioloajat</h3>
+            {fields.map(
+              (
+                item: Partial<ArrayField<Record<string, TimeSpanFormFormat>>>,
+                index
+              ) => (
+                <TimeSpan
+                  item={item}
+                  resourceStateOptions={resourceStateOptions}
+                  setValue={setValue}
+                  register={register}
+                  key={`time-span-${item.id || index}`}
+                  index={index}
+                  remove={remove}
+                />
+              )
+            )}
+          </div>
+          <SecondaryButton
+            dataTest="add-new-time-span-button"
+            onClick={(): void => append({})}
+            className="opening-period-final-action-button add-new-time-span-button">
```

Comment:
This className `opening-period-final-action-button` might need some renaming. Originally it was only applied for the last final action buttons. But now (and actually for some time already) it's also applied in this '+ Lisää aukioloaika' button. And Im not sure if '+ Lisää aukiolaika' is fit for the 'final action button' 😅 . Not in the scope of this PR though, we can do it later.

Reply:
Ok, we can figure out something else for this 👍  Or I can remove the class

Reply:
Removed the class